src/Vatras/Lang/ADT/Merge.agda

```diff
@@ -0,0 +1,133 @@
+open import Vatras.Framework.Definitions
+module Vatras.Lang.ADT.Merge
+  (V : 𝕍)
+  (_+ᵥ_ : ∀ {A} → V A → V A → V A)
+  where
+open import Data.Bool using (Bool; if_then_else_; true; false)
+open import Data.Bool.Properties using (if-float)
+open import Relation.Binary.PropositionalEquality using (refl; _≡_; _≗_)
+open Relation.Binary.PropositionalEquality.≡-Reasoning
+
+open import Vatras.Framework.Relation.Expression V
+open import Vatras.Util.AuxProofs using (if-cong; if-swap)
+
+import Vatras.Lang.ADT
+module ADT F = Vatras.Lang.ADT F V
+
+module Named (F : 𝔽) where
+  open ADT F
+
+  {-|
+  Merges two ADTs.
+  The resulting ADT denotes all possible compositions of variants, such that configuring the resulting ADT
+  is equivalent to configuring both input ADTs and composing the resulting variants (see ⊕-spec below).
+  This operations inherits all properties of the variant composition (e.g., commutativity, associativity etc).
+  -}
+  _⊕_ : ∀ {A} → ADT A → ADT A → ADT A
+  leaf l          ⊕ leaf r          = leaf (l +ᵥ r)
+  leaf l          ⊕ (E ⟨ el , er ⟩) = E ⟨ leaf l ⊕ el , leaf l ⊕ er ⟩
+  (D ⟨ dl , dr ⟩) ⊕ leaf r          = D ⟨ dl ⊕ leaf r , dr ⊕ leaf r ⟩
+  (D ⟨ dl , dr ⟩) ⊕ (E ⟨ el , er ⟩) = D ⟨ E ⟨ dl ⊕ el , dl ⊕ er ⟩ , E ⟨ dr ⊕ el , dr ⊕ er ⟩ ⟩
```

Comment:
Does the indentation of this function look aligned to you in Emacs? With the fonts I use, `⟨` and `⟩` are half-space characters which makes the `=` seem unaligned.

Reply:
Right, didn't notice Github's redering 😅.

Indeed `Symbola`, the font I use as Unicode fallback isn't a strictly monospace font. I just copied your font setup, so now we should have consistent rendering.